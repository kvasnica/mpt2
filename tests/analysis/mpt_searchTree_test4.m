function mpt_searchTree_test4

% test mpt_searchTree() on controllers with only one region
%
% http://control.ee.ethz.ch/~mpt/hg/mpt-26x/rev/e846869a4787

Double_Integrator
probStruct.N = 1;
probStruct.Tconstraint = 0;
probStruct.P_N = zeros(2);
ctrl = mpt_control(sysStruct, probStruct, struct('verbose', 0));
mbg_assertequal(length(ctrl.Pn), 1);

% without a patch, mpt_searchTree would break with an error
st = mpt_searchTree(ctrl);
mbg_assertequal(st.details.searchTree, [0 0 -1 -1]);
