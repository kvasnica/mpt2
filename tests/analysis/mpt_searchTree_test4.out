
Solution consists of 1 regions

Eliminating redundant control laws
Law 1/1        

Extracting hyperplanes
Removing linear dependent hyperplanes


Algorithm finished at iteration 0
